File: ext/Shared/Types/GameEntityData.lua
class 'GameEntityData'

local m_Logger = Logger("GameEntityData", true)

function GameEntityData:__init(arg)
    self.indexInBlueprint = arg.indexInBlueprint;
    self.instanceId = arg.instanceId;
    self.typeName = arg.typeName;
    self.transform = arg.transform;
    self.aabb = arg.aabb;
end

return GameEntityData

File: WebUI/script/types/GameEntityData.js
class GameEntityData {
    constructor(instanceId, indexInBlueprint, typeName, isSpatial, transform, aabb) {
        this.instanceId = instanceId;
        this.indexInBlueprint = indexInBlueprint;
        this.typeName = typeName;
        this.isSpatial = isSpatial;
        this.transform = transform;
        this.aabb = aabb;
    }

    setFromTable(table) {
        this.instanceId = table.instanceId;
        this.indexInBlueprint = table.indexInBlueprint;
        this.typeName = table.typeName;
        this.isSpatial = table.isSpatial;
        if(table.isSpatial) {
            this.transform = new LinearTransform().setFromTable(table.transform);
            this.aabb = new AABB().setFromTable(table.aabb);
        }
        return this;
    }
}